src/query/functions/temporal/aggregation.go

```diff
@@ -192,3 +220,62 @@ func sumAndCount(values []float64) (float64, float64) {
 
 	return sum, count
 }
+
+func makeQuantileOverTimeFn(scalar float64) aggFunc {
+	return func(values []float64) float64 {
+		valuesSlice := make(valsSlice, 0, len(values))
+		for _, v := range values {
+			valuesSlice = append(values, v)
+		}
+
+		return quantile(scalar, valuesSlice)
+	}
+}
+
+type valsSlice []float64
+
+func (s valsSlice) Len() int {
+	return len(s)
+}
+
+func (s valsSlice) Less(i, j int) bool {
+	if math.IsNaN(s[i]) {
+		return true
+	}
+
+	return s[i] < s[j]
+}
+
+func (s valsSlice) Swap(i, j int) {
+	s[i], s[j] = s[j], s[i]
+}
+
+// qauntile calculates the given quantile of a slice of values.
+//
+// This slice will be sorted.
+// If 'values' has zero elements, NaN is returned.
+// If q<0, -Inf is returned.
+// If q>1, +Inf is returned.
+func quantile(q float64, values valsSlice) float64 {
```

Comment:
Nit: keep same name for the `q` variable throughout rather than having it be `scalar` when you parse it